### PR TITLE
ci: Upload docs only in upstream repo

### PR DESCRIPTION
**Pull request by StefanScherer**
Improve the GH actions pipeline to only run the upload and invalidate cache steps in upstream repo.
Otherwise contributors get an error mail from GitHub when they update their `master` branch.

Example:
Screen Shot 2020-03-13 at 10 54 51 AM

I first thought what's wrong with the pipeline in upstream, but then I saw that the full workflow ran in my fork I updated my master branch:

```
git fetch upstream
git rebase upstream/master
git push
```

### CODE REVIEW

**Comment by StefanScherer** [comment]
For reference, I got the idea from the GitHub Actions Advent Calendar:  https://www.edwardthomson.com/blog/github_actions_13_conditionals.html

**Comment by StefanScherer** [comment]
Thanks, I‘ll update and add a timeout 😅💰

**Comment by StefanScherer** [comment]
I've added a timeout.
And I also found out that we deploy a few files from the repo in the final website. I guess we don't want a downloadable https://docs.docker.com/Jenkinsfile 😅

File: .dockerignore [hunk 1]
```diff
@@ -2,7 +2,12 @@
...
+_site
+CONTRIBUTING.md
 Dockerfile
+Dockerfile.archive
 docker-compose.yml
-_site
 Gemfile.lock
+Jenkinsfile
+LICENSE
+README.md
```
Comment (thaJeztah) [inline .dockerignore]: These may actually be needed, as they contain the copyright / licensing info (which should generally be distributed with each copy)

Comment (StefanScherer) [reply .dockerignore]: Good point. I also thought about that. That's what code reviews are good for.
I've updated the commit and keep these two files in the `_site` folder.

**Pull request merged**